async-http-client-backend/zio/src/test/scala/sttp/client/asynchttpclient/zio/SttpBackendStubZioTests.scala

```diff
@@ -26,4 +28,29 @@ class SttpBackendStubZioTests extends AnyFlatSpec with Matchers with ScalaFuture
     runtime.unsafeRun(r).body shouldBe Right("c")
     runtime.unsafeRun(r).body shouldBe Right("a")
   }
+
+  it should "return given web socket response" in {
+    val monad = new RIOMonadAsyncError[Any]
+    val frame1 = WebSocketFrame.text("initial frame")
+    val sentFrame = WebSocketFrame.text("sent frame")
+    val webSocketStub =
+      WebSocketStub
+        .respondWithMessages(List(frame1))(monad)
+        .respondUsing(frame => List(frame.asInstanceOf[WebSocketFrame.Text]))
```

Comment:
Or maybe it would be more practical to accept a `PartialFunction` here, allowing to write:

```
.respondUsing {
  case tm: WebSocketFrame.Text => tm 
}
```
?

Reply:
```
Is
.respondUsing {
  case t: WebSocketFrame.Text => List(t)
  case _                      => List.empty
}
```
that bad?

Reply:
yeah, that's ok :)